### ISSUE TITLE
Poor performance after training on custom data

### ISSUE THREAD

**Issue by amir2628** [JaidedAI/EasyOCR#1012]
Greetings! 👋🏼 

* Can you please give some insights to how is that the accuracy can be as good as `+90%` with really low `validation loss` like `<0.01`, but when using the trained model in production the extracted text is just nonsense and not even close to the actual text in the image? :confused:

* I saw some comments to other issues like this that perhaps using a dataset close to your domain would help, which I used similar images for both training, validation, and inference, but still no changes. ❌ : 🙅🏼‍♂️ 

* Also if I want to just extract text, would it really matter what kind of text would be in the images for training and inference? 🤔 

* Moreover, if you just train one model for let's say 30000 iterations, get the `best_accuracy.pth` and train it again for another 30000 iterations, would it ultimately make the model better? :suspect:  

* In conclusion, I would like to know all of your opinions (specially from the contributors of this repository since they know better what they developed) on why the performance in inference is worse than what it shows in the training process. 🤝 

* If it helps to provide you with anything, let me know. 🗒️ 

Have a good one!

**Comment by MdotO** [comment]
I assume u r training just the recognition model, not detection model. 
Just some points from my side( I think number 2 maybe the most pertinent)
- Please reconfirm whether indeed the training, validation and real life application images are indeed the same, including the **preprocessing** you do on those
- Regarding the gibberish/ nonsense output from the model during inference, I suspect that there maybe a code issue and that the model either doesnt load weights OR reinitiqalizes them to random after loading. A way to confirm that , without looking into code, is to **run inference code** ,same you do for real life application images, on the train/val images again individually and see what the accuracy/output is . I think this may most likely be your case
- If you train for 30000 itr, then retrain with best weights again for 30000 itrs , it wont necessarily mean you end up with better weights

**Comment by IAchraf** [comment]
Hello,
I am in the same problem, I have trained a model with a dataset containing 37500 images, and after training( for 30000 itr), I got 3 .pth files 'custom_example.pth', 'best_accuracy_pth', and 'best_norm_ED.pth'.
My images are only images with some specific characters, like this:
![screenshot_2](https://github.com/JaidedAI/EasyOCR/assets/94184715/78c6bc07-5fd7-4e52-aafe-4793475e3f94)
![screenshot_0](https://github.com/JaidedAI/EasyOCR/assets/94184715/d7bfed10-5b85-41f8-be5e-b74efdb2c3fc)
![screenshot_1](https://github.com/JaidedAI/EasyOCR/assets/94184715/709dd8e8-f013-44c4-90be-16c2022bdb41)
and I did the same thing as this, with every pth file I got :
[https://github.com/JaidedAI/EasyOCR/blob/master/custom_model.md#how-to-use-your-custom-model](url)
**BUT the results were nonsense**

I would appreciate it if you could give me some advice.

**Comment by EivindKjosbakken** [comment]
I think a typical issue that happens when training on custom data, is that the dataset is not representative enough, meaning the model gets overfitted to your specific dataset. There are some ways this can be solved, for example fine-tuning less (lower learning rate/using fewer training iterations), or have a more diverse custom dataset. I made a story on creating a synthetic dataset on TowardsAI here: [https://pub.towardsai.net/how-to-make-a-synthesized-dataset-to-fine-tune-your-ocr-3573f1a7e08b](https://pub.towardsai.net/how-to-make-a-synthesized-dataset-to-fine-tune-your-ocr-3573f1a7e08b) , which could potentially be part of a solution to your problem. Additionally, if your model is giving complete nonsense responses, I would guess it is an issue in the code. I hope this helps:)

**Comment by KAKkkdkkakkda** [comment]
has anyone been able to resolve this issue?

**Comment by KAKkkdkkakkda** [comment]
I actually think I found the cause. You need to place characters and symbols in your custom_example.yaml in the same order as in your training.yaml config you used for fine tuning. It should look like this: numbers, symbols, letters (thats the original order of characters inside your training config, since you add a new language to custom_example.yaml you mess up the order)
